### ISSUE TITLE
[fvt]2.12.3 xcatprobe image -n node -d need to do error switch if xdsh diskless node failed

### ISSUE THREAD

**Issue by junxiawang** [xcat2/xcat-core#1731]
env:all os
build:
c910f03c17k27:/opt/xcat/share/xcat/tools/autotest/result# lsdef -v
lsdef - Version 2.12.2 (git commit 4b3b4f335db70103098931538a6db2be50bbe0d9, built Thu Aug 18 04:06:02 EDT 2016)
root@c910f03c17k27:/opt/xcat/share/xcat/tools/autotest/result# 
How to reproduce:
```

root@c910f03c17k27:/install/netboot# xcatprobe image -n c910f03c17k29 -d
---- Gathering information from all diskless pingable compute nodes ----                                          
c910f03c17k29: @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
c910f03c17k29: @    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
c910f03c17k29: @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
c910f03c17k29: IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
c910f03c17k29: Someone could be eavesdropping on you right now (man-in-the-middle attack)!
c910f03c17k29: It is also possible that a host key has just been changed.
c910f03c17k29: The fingerprint for the ECDSA key sent by the remote host is
c910f03c17k29: SHA256:DtUGbdDEI383AJoW+5B0QVl8gdF9tXiqEmS+Ad0zDGw.
c910f03c17k29: Please contact your system administrator.
c910f03c17k29: Add correct host key in /root/.ssh/known_hosts to get rid of this message.
c910f03c17k29: Offending ECDSA key in /root/.ssh/known_hosts:2
c910f03c17k29:   remove with:
c910f03c17k29:   ssh-keygen -f "/root/.ssh/known_hosts" -R c910f03c17k29
c910f03c17k29: Password authentication is disabled to avoid man-in-the-middle attacks.
c910f03c17k29: Keyboard-interactive authentication is disabled to avoid man-in-the-middle attacks.

c910f03c17k29: Permission denied (publickey,password).

Use of uninitialized value in string eq at /opt/xcat/probe//subcmds/image line 269.
Use of uninitialized value in concatenation (.) or string at /opt/xcat/probe//subcmds/image line 282.
c910f03c17k29: Unmatched os image name or image UUID.                                                             [FAIL]
         Defined:   name = ubuntu16.04.1-ppc64el-netboot-compute uuid = d34f9f09-827a-4f3e-8182-21252ed698b6
         Installed: name =  uuid = N/A
root@c910f03c17k27:/install/netboot# xdsh c910f03c17k29 date
c910f03c17k29: @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
---------------> xdsh node failed need do error switch here

```

**Comment by whowutwut** [comment]
@cxhong  is this similar to the ssh key issue in known_hosts?

**Comment by gurevichmark** [comment]
Fixed by pull request #1741

**Comment by gurevichmark** [comment]
@junxiawang Can you please verify this issue is resolved now.
It was fixed by pull request #1741
